From: Ido Schimmel <idosch@mellanox.com>
Date: Mon, 17 Feb 2020 16:29:36 +0200
Subject: selftests: mlxsw: Remove deprecated test
Patch-mainline: v5.7-rc1
Git-commit: 745a7ea72dc25ffe8c89a2a8d19022243dc8dadc
References: bsc#1176774

The addition of a VLAN on a bridge slave prompts the driver to have the
local port in question join the FID corresponding to this VLAN.

Before recent changes, the operation of joining the FID would also mean
that the driver would enable VXLAN tunneling if a VXLAN device was also
member in the VLAN. In case the configuration of the VXLAN tunnel was
not supported, an extack error would be returned.

Since the operation of joining the FID no longer means that VXLAN
tunneling is potentially enabled, the test is no longer relevant. Remove
it.

Signed-off-by: Ido Schimmel <idosch@mellanox.com>
Acked-by: Jiri Pirko <jiri@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 tools/testing/selftests/drivers/net/mlxsw/extack.sh |   30 --------------------
 tools/testing/selftests/drivers/net/mlxsw/vxlan.sh  |   15 ----------
 2 files changed, 45 deletions(-)

--- a/tools/testing/selftests/drivers/net/mlxsw/extack.sh
+++ b/tools/testing/selftests/drivers/net/mlxsw/extack.sh
@@ -8,7 +8,6 @@ lib_dir=$(dirname $0)/../../../net/forwa
 ALL_TESTS="
 	netdev_pre_up_test
 	vxlan_vlan_add_test
-	port_vlan_add_test
 "
 NUM_NETIFS=2
 source $lib_dir/lib.sh
@@ -104,35 +103,6 @@ vxlan_vlan_add_test()
 
 	ip link del dev vx1
 	ip link del dev br1
-}
-
-port_vlan_add_test()
-{
-	RET=0
-
-	ip link add name br1 up type bridge vlan_filtering 1 mcast_snooping 0
-
-	# Unsupported configuration: mlxsw demands VXLAN with "noudpcsum".
-	ip link add name vx1 up type vxlan id 1000 \
-		local 192.0.2.17 remote 192.0.2.18 \
-		dstport 4789 tos inherit ttl 100
-
-	ip link set dev $swp1 master br1
-	check_err $?
-
-	bridge vlan del dev $swp1 vid 1
-
-	ip link set dev vx1 master br1
-	check_err $?
-
-	bridge vlan add dev $swp1 vid 1 pvid untagged 2>&1 >/dev/null \
-		| grep -q mlxsw_spectrum
-	check_err $?
-
-	log_test "extack - map VLAN at port"
-
-	ip link del dev vx1
-	ip link del dev br1
 }
 
 trap cleanup EXIT
--- a/tools/testing/selftests/drivers/net/mlxsw/vxlan.sh
+++ b/tools/testing/selftests/drivers/net/mlxsw/vxlan.sh
@@ -854,21 +854,6 @@ sanitization_vlan_aware_test()
 	bridge vlan del vid 10 dev vxlan20
 	bridge vlan add vid 20 dev vxlan20 pvid untagged
 
-	# Test that offloading of an unsupported tunnel fails when it is
-	# triggered by addition of VLAN to a local port
-	RET=0
-
-	# TOS must be set to inherit
-	ip link set dev vxlan10 type vxlan tos 42
-
-	ip link set dev $swp1 master br0
-	bridge vlan add vid 10 dev $swp1 &> /dev/null
-	check_fail $?
-
-	log_test "vlan-aware - failed vlan addition to a local port"
-
-	ip link set dev vxlan10 type vxlan tos inherit
-
 	ip link del dev vxlan20
 	ip link del dev vxlan10
 	ip link del dev br0
